src/index.js

```diff
@@ -4,67 +4,74 @@ const params = new URLSearchParams(location.search)
 
 var manifest = params.get('m')
 var canvas = params.get('c')
+var config = {}
+var annotations = {}
 
 if (manifest && canvas) {
   console.log ('detected URL params')
 } else {
   console.log ('detected InvenioRDM preview URL')
-
-  //compute manifest URL from InvenioRDM preview URL
-  manifest = location.href.split('?')[0]
-  manifest=manifest.replace('records/','api/iiif/record:');
-  manifest=manifest.replace(/preview.+$/, 'manifest');
-
-  canvas = location.href.split('?')[0]
-  canvas=canvas.replace('records/','api/iiif/record:');
-  canvas=canvas.replace('/preview/', '/canvas/');
-
-  if (params.get('preview')) {
-    manifest = manifest.replace('record:','draft:');
-    canvas = canvas.replace('record:','draft:');
-  }
-
+  manifest = document.getElementById('m3-dist').getAttribute('data-manifest');
+  canvas = document.getElementById('m3-dist').getAttribute('data-canvas');
+  config = JSON.parse(document.getElementById('m3-dist').getAttribute('data-config'));
+  annotations = JSON.parse(document.getElementById('m3-dist').getAttribute('data-annotations'));
```

Comment:
- Shouldn't these be inside `initMirador` and also use the `id` parameter for looking-up the element?
  - We should still keep the behaviour though that parses the manifest and canvas from the `URLSearchParams`
- We should use [`el.dataset`](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) to get the data attributes

```suggestion
  const el = document.getElementById(id)
  manifest = el.dataset.manifest;
  canvas = el.dataset.canvas;
  config = JSON.parse(el.dataset.config);
  annotations = JSON.parse(el.dataset.annotations);
```